sessionctx/variable/varsutil.go

```diff
@@ -574,3 +574,41 @@ var GAFunction4ExpressionIndex = map[string]struct{}{
 	ast.JSONKeys:          {},
 	ast.JSONLength:        {},
 }
+
+var analyzeSkipAllowedTypes = map[string]struct{}{
+	"json":       {},
+	"text":       {},
+	"mediumtext": {},
+	"longtext":   {},
+	"blob":       {},
+	"mediumblob": {},
+	"longblob":   {},
+}
+
+// ValidAnalyzeSkipColumnTypes makes validation for tidb_analyze_skip_column_types.
+func ValidAnalyzeSkipColumnTypes(val string) (string, error) {
```

Comment:
Why do we add this validation? It seems that all types should be allowed to skip.

Reply:
For caution, we constrain the column types which can be skipped to `json,blob,mediumblob,longblob,text,mediumntext,longtext`.

Reply:
Skipping a int-type column doesn't have much benefit but accidentally `set tidb_analyze_skip_column_types = 'int'` may bring some trouble.